src/lib/device_tree/src/lib.rs

```diff
@@ -109,10 +109,10 @@ impl Path {
 /// In-memory reader for Flattened Device Tree format.
 ///
 /// Does not perform any sanity checks.
-pub struct FdtReader<'a> {
-    _mem_reservation_block: SectionReader<'a>,
-    struct_block: SectionReader<'a>,
-    strings_block: SectionReader<'a>,
+pub struct FdtReader<'a, D: Driver> {
+    _mem_reservation_block: SectionReader<'a, D>,
+    struct_block: SectionReader<'a, D>,
+    strings_block: SectionReader<'a, D>,
 }
 
 fn cursor_u32(drv: &dyn Driver, cursor: &mut usize) -> Result<u32> {
```

Comment:
Would it be in the spirit of your PR to change all dyn to impl?

Reply:
Yes, changed all `dyn` uses inside device_tree crate to `impl`, thanks!